webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -278,4 +303,161 @@ describe('TasksComponent', () => {
       expect(component.listTrackBy(0, false)).to.equal(undefined);
     });
   });
+
+  describe('lineage and breadcrumbs', () => {
+    const bettysContactDoc = {
+      _id: 'user',
+      parent: {
+        _id: 'parent',
+        name: 'CHW Bettys Area',
+      },
+    };
+    const taskDocs = [
+      { _id: '1', emission: { _id: 'e1', dueDate: '2020-10-20' }, forId: 'a', owner: 'a' },
+      { _id: '2', emission: { _id: 'e2', dueDate: '2020-10-20' }, forId: 'b', owner: 'b' },
+    ];
+    const taskLineages = [
+      {
+        _id: 'a', lineage: [
+          {name: 'Amy Johnsons Household'},
+          {name: 'St Elmos Concession'},
+          {name: 'Chattanooga Village'},
+          {name: 'CHW Bettys Area'},
+        ]
+      },
+      {
+        _id: 'b', lineage: [
+          {name: 'Amy Johnsons Household'},
+          {name: 'St Elmos Concession'},
+          {name: 'Chattanooga Village'},
+        ]
+      }
+    ];
+
+    it('it should not change the reports lineages if the connected user is online only', async() => {
+      const expectedTasks =
+        [
+          {
+            _id: 'e1',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+              'CHW Bettys Area'
+            ],
+            overdue: true,
+            owner: 'a',
+          },
+          {
+            _id: 'e2',
+            date: new Date('2020-10-20'),
+            dueDate: '2020-10-20',
+            lineage: [
+              'Amy Johnsons Household',
+              'St Elmos Concession',
+              'Chattanooga Village',
+            ],
+            overdue: true,
+            owner: 'b',
+          }
+        ];
+      userContactService.get.resolves(bettysContactDoc);
+      sessionService.isOnlineOnly.returns(true);
+      rulesEngineService.fetchTaskDocsForAllContacts.resolves(taskDocs);
+      lineageModelGeneratorService.reportSubjects.resolves(taskLineages);
+      await new Promise(resolve => {
+        sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+        getComponent();
+      });
+      expect(component.currentLevel).to.be.undefined;
+      expect((<any>TasksActions.prototype.setTasksList).args).to.deep.equal([[expectedTasks]]);
+    });
+
+    it('it should not change the reports lineages if the connected user is offline with unrelated lineage', async() => {
```

Comment:
Targets! no reports :)

And space after async as it's reserved word.
```suggestion
    it('should not alter tasks lineage if user is offline with unrelated lineage', async () => {
```

Reply:
These things are usually caught in a self-code review and that would have saved time